src/backends/r1cs/mod.rs

```diff
@@ -379,13 +390,24 @@ where
         Ok(())
     }
 
-    fn compute_var(
-        &self,
-        env: &mut crate::witness::WitnessEnv<Self::Field>,
-        var: Self::CellVar,
-    ) -> crate::error::Result<Self::Field> {
-        let val = self.witness_vars.get(var.index).unwrap();
-        self.compute_val(env, val, var.index)
+    fn compute_var(&self, env: &mut crate::witness::WitnessEnv<Self::Field>, var: Self::CellVar) -> Result<Self::Field> {
+        match var {
+            R1csCellVar::Var(wv) => {
+                self.compute_val(env, self.witness_vars.get(wv.index).unwrap(), wv.index)
+            },
+            // if it's a linear combination, we need to evaluate each term
+            R1csCellVar::LinearCombination(lc) => {
+                let mut val = lc.constant;
+
+                for (wv, factor) in lc.terms {
+                    let wv_val = self.witness_vars.get(wv.index).unwrap();
+                    let calc = self.compute_val(env, wv_val, wv.index)? * factor;
+                    val += calc;
+                }
+                
+                Ok(val)
```

Comment:
mmm, I'm wondering if you should panic if it's a linear combination here. (Or if you refactor the code as I explained, then we should always expect a single term (that's not the first variable) and a 0 constant

Reply:
Refactored